src/foam/nanos/auth/SetUserServiceProviderJunctionRuleAction.js

```diff
@@ -73,7 +73,7 @@ foam.CLASS({
             serviceProvider.removeSpid(x, user);
 
             // get grantpath and create/reput grantpath
-            serviceProvider.setupSpid(x, user);
+            serviceProvider.setupSpid(x, user, false);
```

Comment:
Maybe can check if the user.spid is same as the service provider before re-setting up the user spid?

Reply:
Yes, looking a performance, this seems to be an unnecessary re-assignment when the spid has not changed.